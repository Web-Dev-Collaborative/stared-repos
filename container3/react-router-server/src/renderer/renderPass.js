import React from 'react';
import { renderToString, renderToStaticMarkup } from 'react-dom/server';
import AsyncRenderer from '../components/AsyncRenderer';
import removeDuplicateModules from '../utils/removeDuplicateModules';

const renderPass = (context, element, staticMarkup = false) => {
  context.callback = () => {
    if (context.finishedLoadingModules && !context.statesRenderPass) {
      context.statesRenderPass = true;
      context.renderResult = renderPass(context, element, staticMarkup);
      if (context.fetchingStates <= 0 && context.modulesLoading <= 0) {
        context.resolve({
          html: context.renderResult,
          state: context.fetchStateResults === undefined ? null : context.fetchStateResults,
          modules: context.modules === undefined ? null : removeDuplicateModules(context.modules)
        });
      }
    } else if (context.finishedLoadingModules && context.statesRenderPass || !context.hasModules) {
      context.renderResult = renderPass(context, element, staticMarkup);
      if (context.fetchingStates <= 0 && context.modulesLoading <= 0) {
        context.resolve({
          html: context.renderResult,
          state: context.fetchStateResults === undefined ? null : context.fetchStateResults,
          modules: context.modules === undefined ? null : removeDuplicateModules(context.modules)
        });
      }
    }
  };

  let component = (
    <AsyncRenderer context={context}>
      {element}
    </AsyncRenderer>
  );

  let result;
  try {
    if (staticMarkup) {
      result = renderToStaticMarkup(component);
    } else {
      result = renderToString(component);
    }
  } catch (e) {
    return context.reject(e);
  }

  if (!context.hasModules && !context.hasStates) {
    context.resolve({
      html: result,
      state: context.fetchStateResults === undefined ? null : context.fetchStateResults,
      modules: context.modules === undefined ? null : removeDuplicateModules(context.modules)
    });
  }
  return result;
};

export default renderPass;
