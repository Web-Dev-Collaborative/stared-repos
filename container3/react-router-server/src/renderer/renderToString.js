import renderPass from './renderPass';
import { clear } from '../module/cache';

export default (element) => {
  return new Promise((resolve, reject) => {
    clear();
    const context = {
      resolve, reject,
      modulesLoading: 0,
      fetchingStates: 0
    };
    renderPass(context, element);
  });
};
