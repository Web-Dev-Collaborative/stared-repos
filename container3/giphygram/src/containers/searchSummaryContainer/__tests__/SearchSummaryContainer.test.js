import React from 'react';
import renderer from 'react-test-renderer';
import SearchSummaryContainerConnected, { SearchSummaryContainer } from '../SearchSummaryContainer';

jest.mock(
  '../../../components/searchSummary/SearchSummary',
  () => ({
    SearchSummary: 'SearchSummary',
  }),
);

jest.mock('react-redux', () => ({
  connect: mapStateToProps => (Component) => {
    const state = {
      searchResults: {
        data: [],
        pagination: {
          total_count: 100,
        },
        meta: {},
        isLoading: false,
        isFetchingMore: false,
        error: null,
      },
    };
    const propsFromState = mapStateToProps(state);
    return () => <Component {...propsFromState} />;
  },
}));

describe('SearchSummaryContainer', () => {
  it('should be rendered correctly by default', () => {
    const tree = renderer
      .create(<SearchSummaryContainer />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should be rendered and connected correctly by default', () => {
    const tree = renderer
      .create((
        <SearchSummaryContainerConnected />
      ))
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should be rendered correctly with specified total parameter', () => {
    const tree = renderer
      .create(<SearchSummaryContainer total={42} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
