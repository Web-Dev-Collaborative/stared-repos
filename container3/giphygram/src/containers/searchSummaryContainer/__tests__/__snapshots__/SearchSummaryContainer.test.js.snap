// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSummaryContainer should be rendered and connected correctly by default 1`] = `
<SearchSummary
  total={100}
/>
`;

exports[`SearchSummaryContainer should be rendered correctly by default 1`] = `
<SearchSummary
  total={null}
/>
`;

exports[`SearchSummaryContainer should be rendered correctly with specified total parameter 1`] = `
<SearchSummary
  total={42}
/>
`;
